src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,176 @@
+        String out = "Hello! I'm Duke\n"
+                + "What can I do for you?\n";
+        System.out.println(out);
+    }
+
+    /**
+     * Show unknown command messages
+     */
+    public void showUnknownMessage() {
+        System.out.println("I'm sorry, "
+                + "but I don't know what that means :-(");
+    }
+
+    /**
+     * Show add messages
+     *
+     * @param task Task added
+     * @param size Size of TaskList
+     */
+    public void showAddMessage(Task task, int size) {
+        String plural = size == 1
+                ? "task"
+                : "tasks";
+        String out = "Got it. I've added this task:\n "
+                + task + "\nNow you have "
+                + size + " " + plural + " in the list.";
+        System.out.println(out);
+    }
+
+    /**
+     * Show delete messages.
+     *
+     * @param task Task that is going to be deleted
+     * @param size Size of TaskList
+     */
+    public void showDeleteMessage(Task task, int size) {
+        String info = task.toString();
+        String plural = size - 1 <= 1
+                ? "task"
+                : "tasks";
+        String out = "Noted. I've removed this task:\n  "
+                 + info + "\n Now you have " + size
+                + " " + plural + " in the list";
+        System.out.println(out);
+    }
+
+    /**
+     * Show exit messages when the program is going to end.
+     */
+    public void showExitMessage() {
+        System.out.println("Bye. Hope to see you again soon!");
+    }
+
+    /**
+     * Show header of list command.
+     */
+    public void showList() {
+        System.out.println("Your List :");
+    }
+
+    /**
+     * Show the details of task.
+     *
+     * @param index Index of task
+     * @param task Task to print
+     */
+    public void showTask(int index, Task task) {
+        System.out.println(index + "." + task.toString());
+    }
+
+    /**
+     * Show mark messages.
+     *
+     * @param task Task marked as done
+     */
+    public void showMarkMessage(Task task) {
+        String out = "Nice! I've marked this task as done:\n  " + task.toString();
+        System.out.println(out);
+    }
+
+    /**
+     * Show unmark messages.
+     *
+     * @param task Task unMarked as done.
+     */
+    public void showUnmarkMessage(Task task) {
+        String out = "OK, I've marked this task as not done yet:\n  " + task.toString();
+        System.out.println(out);
+    }
+
+    /**
+     * Show find message when tasks are found.
+     */
+    public void showFindMessage() {
+        System.out.println("Here are the matching tasks in your list:\n");
+    }
+
+    /**
+     * Show find message when no task is found.
+     */
+    public void showFindEmptyMessage() {
+        System.out.println("There are no matching task in your list\n");
+    }
+
+    /**
+     * Show error messages.
+     *
+     * @param message Error message
+     */
+    public void showLoadingError(String message) {
```

Comment:
Would it perhaps be a better idea to pass in the Exception object as an argument to this method, maybe to make the method more descriptive?